process/interceptors/processor/txInterceptorProcessor.go

```diff
@@ -38,10 +41,9 @@ func (txip *TxInterceptorProcessor) Validate(data process.InterceptedData) error
 		return process.ErrWrongTypeAssertion
 	}
 
-	//TODO change the IsTxValidForProcessing to output error
-	isTxValid := txip.txValidator.IsTxValidForProcessing(interceptedTx)
-	if !isTxValid {
-		return process.ErrTxNotValid
+	errTxValidation := txip.txValidator.CheckTxValidity(interceptedTx)
+	if errTxValidation != nil {
+		return errors.New(fmt.Sprintf("%s: %s", process.ErrTxNotValid.Error(), errTxValidation))
```

Comment:
return the err and that's it. Change errTxValidation to err.

Reply:
changed. Now i return only the error